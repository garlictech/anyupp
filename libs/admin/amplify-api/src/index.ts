export * from './lib/models/index.js';
