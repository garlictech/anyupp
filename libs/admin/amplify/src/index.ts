export * from './lib/api';

import * as Mutations from './lib/documents/mutations';
import * as Queries from './lib/documents/queries';
import * as Subscriptions from './lib/documents/subscriptions';

import  awsmobile  from './lib/aws-exports';

export { Mutations, Queries, Subscriptions, awsmobile}