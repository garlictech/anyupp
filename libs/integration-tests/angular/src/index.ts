export * from './lib/integration-tests-angular.module';
