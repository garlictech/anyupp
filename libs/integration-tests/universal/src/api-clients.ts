import {
  awsConfig,
  getCrudSdkForIAM,
  getCrudSdkForUserPool,
} from '@bgap/crud-gql/api';
import {
  getAnyuppSdkForIAM,
  getAnyuppSdkForUserPool,
} from '@bgap/anyupp-gql/api';
import { Auth } from 'aws-amplify';
import { from } from 'rxjs';
import { map } from 'rxjs/operators';

Auth.configure({
  ...awsConfig,
  // See: https://github.com/aws-amplify/amplify-js/issues/6552#issuecomment-682259256
  authenticationFlowType: 'USER_PASSWORD_AUTH',
  aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
});

export const createAuthenticatedCrudSdk = (
  userName: string,
  password: string,
) => from(Auth.signIn(userName, password)).pipe(map(getCrudSdkForUserPool));

export const createAuthenticatedAnyuppSdk = (
  userName: string,
  password: string,
) => from(Auth.signIn(userName, password)).pipe(map(getAnyuppSdkForUserPool));

export const createIamCrudSdk = () =>
  getCrudSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );

export const createIamAnyuppSdk = () =>
  getAnyuppSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );
