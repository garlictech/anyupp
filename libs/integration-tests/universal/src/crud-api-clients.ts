import {
  awsConfig,
  getCrudSdkForIAM,
  getCrudSdkForUserPool,
} from '@bgap/crud-gql/api';
import { Auth } from 'aws-amplify';
import { from, Observable } from 'rxjs';
import { map, mapTo } from 'rxjs/operators';

export const createAuthenticatedCrudSdk = (
  userName: string,
  password: string,
) => {
  Auth.configure({
    ...awsConfig,
    // See: https://github.com/aws-amplify/amplify-js/issues/6552#issuecomment-682259256
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });

  return from(Auth.signIn(userName, password)).pipe(map(getCrudSdkForUserPool));
};

export const createIamSdk = () =>
  getCrudSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );
