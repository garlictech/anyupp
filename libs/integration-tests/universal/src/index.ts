export { testAdminUsername, testAdminUserPassword } from './lib/common';
