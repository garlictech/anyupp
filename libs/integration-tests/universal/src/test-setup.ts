(global as any).WebSocket = require('ws');
require('es6-promise').polyfill();
require('isomorphic-fetch');
