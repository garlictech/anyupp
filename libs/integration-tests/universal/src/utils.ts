import * as R from 'ramda';
import { pipe } from 'fp-ts/lib/function';

export const ES_DELAY = 3000; //ms
export const DYNAMO_DELAY = 1000; //ms

// Replace UUID-s with a static string for jest matchers
export const maskV4UuidIds = <T>(data: T) =>
  pipe(
    data,
    JSON.stringify,
    R.replace(
      /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/g,
      'MASKED ID',
    ),
    (d: string) => JSON.parse(d) as T,
  );
