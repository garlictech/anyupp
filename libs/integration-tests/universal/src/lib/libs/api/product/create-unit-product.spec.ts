import {
  AmplifyApi,
  AmplifyApiQueryDocuments,
} from 'libs/admin/amplify-api/src';
import { validateUnitProduct } from 'libs/shared/data-validators/src';
import { IProduct } from 'libs/shared/types/src';
import { Observable, from } from 'rxjs';
import { map, switchMap } from 'rxjs/operators';

import { AppsyncApi } from '@bgap/api/graphql/schema';
import {
  appsyncGraphQlClient,
  AuthenticatdGraphQlClientWithUserId,
  createAuthenticatedAppsyncGraphQlClient,
  executeMutation,
  executeQuery,
  GraphqlApiClient,
} from '@bgap/shared/graphql/api-client';

import {
  testAdminUsername,
  testAdminUserPassword,
  unitProductSeed,
} from '../../../fixtures';

const input: AppsyncApi.CreateUnitProductMutationVariables = {
  input: unitProductSeed.unitProduct_01,
} as any;

describe('CreateUnitProduct tests', () => {
  it('should require authentication to access', done => {
    return executeMutation(appsyncGraphQlClient)<
      AppsyncApi.CreateUnitProductMutation
    >(AppsyncApi.CreateUnitProduct, input).subscribe({
      error(e) {
        expect(e).toMatchSnapshot();
        done();
      },
    });
  }, 25000);

  describe.only('with authenticated user', () => {
    let authHelper: AuthenticatdGraphQlClientWithUserId;

    beforeAll(async () => {
      authHelper = await createAuthenticatedAppsyncGraphQlClient(
        testAdminUsername,
        testAdminUserPassword,
      ).toPromise();
      console.warn(authHelper.userAttributes);
    });
    // let authenticatedApsyncGraphQlClient;
    // beforeAll(async () => {
    //   authenticatedApsyncGraphQlClient = createAuthenticatedAppsyncGraphQlClient()
    //   await combineLatest([
    //     // CleanUP
    //     // deleteTestCart(),
    //     // deleteTestCart(cartWithNotExistingUNIT),
    //   ])
    //     .pipe(
    //       switchMap(() =>
    //         // Seeding
    //         combineLatest([
    //           // createTestCart(),
    //           // createTestCart({
    //           //   id: cartWithNotExistingUNIT,
    //           //   unitId: unitSeed.unitId_NotExisting,
    //           // }),
    //         ]),
    //       ),
    //     )
    //     .toPromise();
    // }

    // it('should .... with authenticated user', done => {
    //   // return executeMutation(authHelper.graphQlClient)<
    //   //   AppsyncApi.CreateUnitProductMutation
    //   // >(AppsyncApi.CreateUnitProduct, input)
    //   from(productRequestHandler.createUnitProduct(amplifyGraphQlClient)(input))
    //     .pipe(
    //       pipeDebug('### UNITPRODUCT CREATE RESULT'),
    //       // map(result => result.createUnitProduct),
    //       // switchMap(product =>
    //       //   // getUnitProduct(amplifyGraphQlClient, product.id),
    //       // ),
    //     )
    //     .subscribe({
    //       next(result) {
    //         console.log(
    //           '### ~ file: create-unit-product.spec.ts ~ line 69 ~ next ~ result',
    //           JSON.stringify(result, undefined, 2),
    //         );
    //         expect(result).toMatchSnapshot();

    //         done();
    //       },
    //     });
    // });
  });
});

// const getUnitProduct = (
//   amplifyApiClient: GraphqlApiClient,
//   productId: string,
// ): Observable<IProduct> => {
//   return executeQuery(amplifyApiClient)<AmplifyApi.GetUnitProductQuery>(
//     AmplifyApiQueryDocuments.getUnitProduct,
//     { id: productId },
//   ).pipe(
//     map(product => product.getUnitProduct),
//     switchMap(validateUnitProduct),
//   );
// };
