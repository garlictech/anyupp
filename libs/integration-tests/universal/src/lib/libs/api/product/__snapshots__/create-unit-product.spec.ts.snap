// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUnitProduct tests should require authentication to access 1`] = `[Error: GraphQL error: Not Authorized to access createUnitProduct on type Mutation]`;

exports[`CreateUnitProduct tests with authenticated user should .... with authenticated user 1`] = `[Error: GraphQL error: Cannot return null for non-nullable type: 'ID' within parent 'Mutation' (/createUnitProduct)]`;
